### ISSUE TITLE
I cant really get this thing to work, followed the instruction, downloades node but i am getting an error when i try to run it

### ISSUE THREAD

**Issue by Brigadaus** [streamwall/streamwall#123]
opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

Node.js v18.15.0

is this to outdated to run anymore or what?

**Comment by crannofonix** [comment]
I get the same issue on Windows 8.1 but not on Windows 10

**Comment by chromakode** [comment]
Streamwall is overdue an electron / node upgrade. I'm currently busy with other projects, so can't estimate when I'll be able to get to this. I'd welcome a PR that bumps `electron` to the latest version and tests functionality.

**Comment by Brigadaus** [comment]
Hey Chromakode, i understand that this isn't a high priority thing in your life anymore, saw you had some health issues also at some point and the world have moved on from where it was with all the protests in the US back then, and you have more important things to focus on and probably have to earn money doing actual paid work and so on, i'm not that much into this kind of programing, but i feel like it would have been smarter to have ported this to something that wouldn't constantly require all this maintenance for it not to break, even if it was just made as a chrome or firefox app it wouldn't require constant maintenance, anyways that's just my thoughts, also i think the world is going to explode in all kinds of protests the coming year or two, like we see in europe now, ive personally been covering the paris protests lately with thousands of viewers hungering for the multiscreen streams, (which i just do with lots of chrome instances that i use a chrome app that remove all the youtube things from the window)  and even though US has dialed down with the protests for now, the world might blow up soon including in the US so multistreaming will be growing again, and you made the best software to do it with. The fight against the powers that rule us is not over. All the best @chromakode

**Comment by chromakode** [comment]
> it would have been smarter to have ported this to something that wouldn't constantly require all this maintenance for it not to break, even if it was just made as a chrome or firefox app it wouldn't require constant maintenance

Funny you would say that... I originally tried to make Streamwall in Chrome and bumped up against the limitations all browser-based options do. The electron architecture is a big part of what makes Streamwall so effective.

I agree that there's important work to come. I will continue to update Streamwall as time permits.

**Comment by Brigadaus** [comment]
Yeah i was thinking that probably chrome wouldn't be too happy with this, that's why i was thinking about firefox too, i use "Windowed - floating Youtube/every website" on chrome, which puts in 3 options when you hit full screen on any video, one of them "windowed" basically pulls the video out in a separate window consisting of only the video player and you can open as many as you want, which works fine, but theres no attribution and you have to manually scale all the windows. Thank you for taking the time to answer, all the best

**Comment by chromakode** [comment]
I've pushed a lightly tested new version with electron and other dependencies updated: https://github.com/streamwall/streamwall/commit/a581ba2c6f536abbac1d1c1a6326a681b6138c40

This should resolve the above issue. I'll push this version to the release branch after it's had some more testing.

**Comment by Brigadaus** [comment]
Wow, thank you very much!, can confirm it works for me now. Amazing

**Comment by Brigadaus** [comment]
So after some quick testing i see that it's not loading tiktok livestreams, if you put the tiktok url ending with the username of the channel, it loads the last video the user uploaded, not his current livestream , if you then add /live after the username, it doesn't load anything at all.

Also i don't know if this is normal but the streamwall window isn't full 1920x1080, seems to be a little smaller, also tried to start it from a toml file where i specified the resolution 1920x1080, but still isnt full resolution 
![image](https://user-images.githubusercontent.com/16061383/232194263-008014dd-ed89-48ae-b8df-3694f549f72f.png)

**Comment by UnTelevised** [comment]
> > it would have been smarter to have ported this to something that wouldn't constantly require all this maintenance for it not to break, even if it was just made as a chrome or firefox app it wouldn't require constant maintenance
> 
> Funny you would say that... I originally tried to make Streamwall in Chrome and bumped up against the limitations all browser-based options do. The electron architecture is a big part of what makes Streamwall so effective.
> 
> I agree that there's important work to come. I will continue to update Streamwall as time permits.

I am thinking this project would do better to move to nextjs I am looking at the undertaking now. the code is so complex though its taking me some time to fully understand it

**Comment by Brigadaus** [comment]
Nice, i see that you are covering the protests too so you probably have some ideas how it can be improved, really hope to see it, please let me know if you get to releasing something

**Comment by sayhiben** [comment]
> I am thinking this project would do better to move to nextjs 

I'm curious as to what you think this might improve or fix aside from adding another build dependency

**Comment by UnTelevised** [comment]
> > I am thinking this project would do better to move to nextjs
> 
> I'm curious as to what you think this might improve or fix aside from adding another build dependency

Not so much an additional dependency as a rework into a whole new environment. But I am thinking a more web deployable version.
My use case idea is running a media outlet that uses aggregated streams in many ways including from direct broadcast as well as making the studio more accessable on the go. 

However i dont think my idea will work due to how electron functions. I do not believe it can be mae to trigger its window for a web user but I have not had time to look into that, its where I got stuck, it can function fine in next as a locally deployed application, and my crude set up runs a lot smoother than the node version.

**Comment by sayhiben** [comment]
> I do not believe it can be mae to trigger its window for a web user

Oh, what do you mean by "trigger" here? I've been poking at this project again since the college protests picked up, and I'm almost back up to speed with things. There's a lot of cleanup / polish opportunity here, which is why I was curious about your interest in Next.js. I'm not sure it solves any problems in Electron, as you say, but it's a good line of thinking

In my opinion, there's a distribution issue with this project that we can solve by creating multi-platform releases with something like [electron-builder](https://www.electron.build/). I've toyed with Dockerizing the development of this app, but ultimately I think it should be able to be run natively (i.e., without a working Node dev env) so that OBS and other broadcasting software can "just work." Distribution and catching up on tech debt feels like a good place to spend effort here at the moment

It sounds like your use-case is a little different in that you want a website where users can effectively run their own Streamwall, but within the browser. I'd have to give it some more thought in terms of how that could be accomplished, because much of Streamwall depends on using Electron's BrowserView, which is sort of like an iOS or Android WebView. Off-the-cuff, I don't know how that could be done in a traditional browser without running into some significant iframe and CORS limitations
